### DISCUSSION TITLE
preservation of Macros in xlsm template

### DISCUSSION THREAD

**Discussion by aruba67416** [JanMarvin/openxlsx2#791] [Q&A] [answered]
Hello, I have been using openxlsx to read a template .xlsm document which contains macros. After adding data and saving as .xlsm with saveWorkbook, the macros don't work any longer. Does openxlsx2 support existing macros ?

Having searched on this topic, I see that others have ran into the same situation. However, I there is no resolution to be found. I would appreciate guidance from the community.

**Comment by JanMarvin** [comment, marked as answer]
Hi @aruba67416 , yes this should be supported. Please give it a try and open an issue with steps to reproduce it otherwise.
